core/templates/dev/head/domain/skill/ConceptCardBackendApiService.ts

```diff
@@ -52,73 +52,59 @@ oppia.factory('ConceptCardBackendApiService', [
-       * parameter. If something goes wrong while trying to fetch the
-       * concept card, the rejection callback is called instead, if present. The
-       * rejection callback function is passed the error that occurred and the
-       * skill ID.
-       */
-      fetchConceptCard: function(skillId) {
-        return $q(function(resolve, reject) {
-          _fetchConceptCard(skillId, resolve, reject);
-        });
-      },
+    var _getUncachedSkillIds = function(skillIds) {
+      var uncachedSkillIds = [];
+      skillIds.forEach(function(skillId) {
+        if (!_isCached(skillId)) {
+          uncachedSkillIds.push(skillId);
+        }
+      });
+      return uncachedSkillIds;
+    };
 
+    return {
       /**
-       * Behaves in the exact same way as fetchConceptCard (including callback
+       * Behaves in the exact same way as fetchConceptCards (including callback
        * behavior and returning a promise object), except this function will
-       * attempt to see whether the given concept card has already been loaded.
-       * If it has not yet been loaded, it will fetch the concept card from the
-       * backend. If it successfully retrieves the concept card from the
-       * backend, it will store it in the cache to avoid requests from the
-       * backend in further function calls.
+       * attempt to see whether the given concept cards have already been
+       * loaded. If they have not yet been loaded, it will fetch the concept
+       * cards from the backend. If it successfully retrieves the concept cards
+       * from the backend, it will store them in the cache to avoid requests
+       * from the backend in further function calls.
        */
-      loadConceptCard: function(skillId) {
+      loadConceptCards: function(skillIds) {
         return $q(function(resolve, reject) {
-          if (_isCached(skillId)) {
+          var uncachedSkillIds = _getUncachedSkillIds(skillIds);
+          var conceptCards = [];
+          // Concept card partial cache hit case.
+          if (uncachedSkillIds.length !== 0) {
+            _fetchConceptCards(
+              uncachedSkillIds, function(uncachedConceptCards) {
+                skillIds.forEach(function(skillId) {
+                  if (uncachedSkillIds.includes(skillId)) {
+                    conceptCards.push(
+                      uncachedConceptCards[uncachedSkillIds.indexOf(skillId)]);
+                    // Save the fetched conceptCards to avoid future fetches.
+                    _conceptCardCache[skillId] = angular.copy(
+                      uncachedConceptCards[uncachedSkillIds.indexOf(skillId)]);
+                  } else {
+                    conceptCards.push(angular.copy(_conceptCardCache[skillId]));
+                  }
+                });
+                if (resolve) {
+                  resolve(angular.copy(conceptCards));
+                }
+              }, reject);
+          } else {
+            // Concept card complete cache hit case.
```

Comment:
This comment reads a bit cryptically (albeit nicely alliterative :P). Can you elaborate a bit? E.g. "This is the case where all concept cards have already been cached locally." or something like that.

Reply:
Done.